compiler/rustc_session/src/config.rs

```diff
@@ -2089,11 +2089,11 @@ fn select_debuginfo(matches: &getopts::Matches, cg: &CodegenOptions) -> DebugInf
 
 fn parse_assert_incr_state(
     early_dcx: &EarlyDiagCtxt,
-    opt_assertion: &Option<String>,
+    opt_assertion: Option<&str>,
 ) -> Option<IncrementalStateAssertion> {
     match opt_assertion {
-        Some(s) if s.as_str() == "loaded" => Some(IncrementalStateAssertion::Loaded),
-        Some(s) if s.as_str() == "not-loaded" => Some(IncrementalStateAssertion::NotLoaded),
+        Some(s) if s == "loaded" => Some(IncrementalStateAssertion::Loaded),
+        Some(s) if s == "not-loaded" => Some(IncrementalStateAssertion::NotLoaded),
```

Comment:
You can write this without if-guards now because you no longer need to call a function

Reply:
good catch, thx